src/api/resources/sync/views.py

```diff
@@ -181,10 +181,15 @@ def _format_errors(errors):
     return {k: v[0] for k, v in errors.items()}
 
 
-def _get_serialized_record(serializer, record_id):
-    record = get_record(model_class=serializer.Meta.model, record_id=record_id)
+def _get_serialized_record(viewset, record_id):
+    serializer = viewset.serializer_class
+    record = get_record(viewset, record_id)
 
-    data = serialize_revisions(serializer, [record])
+    if isinstance(record, dict):
```

Comment:
Why not have `get_records` return the same data type (list of dicts) for updates and deletes, and avoid this?